nextcord/interactions.py

```diff
@@ -372,12 +378,23 @@ async def delete_original_message(self) -> None:
             Deleted a message that is not yours.
         """
         adapter = async_context.get()
-        await adapter.delete_original_interaction_response(
+        delete_func = adapter.delete_original_interaction_response(
             self.application_id,
             self.token,
             session=self._session,
         )
 
+        if delay is not None:
+            async def inner_call(delay: float = delay):
```

Comment:
Maybe use call_later?

Reply:
This is the code from `InteractionMessage.delete` but repurposed.

I think I'd need the event loop to use call_later? Not sure how it would be changed.

Reply:
I suppose it could be changed to use self._state.loop.call_later but it's not much different

```py
async def inner_call(delay: float = delay):
    try:
        await delete_func
    except HTTPException:
        pass
self._state.loop.call_later(delay, inner_call)
```